src/Nethermind/Nethermind.Core/Extensions/SpecProviderExtensions.cs

```diff
@@ -0,0 +1,36 @@
+//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using Nethermind.Core.Specs;
+
+namespace Nethermind.Core.Extensions
+{
+    public static class SpecProviderExtensions
+    {
+        /// <summary>
+        /// this method is here only for getting spec for 1559.
+        /// Reason of adding is that at sometime we dont know the Timestamp.
+        /// </summary>
+        /// <param name="specProvider"></param>
+        /// <param name="blockNumber"></param>
+        /// <returns>ReleaseSpec that has the values for EIP1559 correct but not the rest.</returns>
+        public static IReleaseSpec GetSpecFor1559(this ISpecProvider specProvider, long blockNumber)
+        {
+            return specProvider.GetSpec(blockNumber);
```

Comment:
so we have still public GetSpec method. I am not convinced if it is the best way to avoid problems in future.

Reply:
Addressed in the latest changes